src/models/spaces/charts/chart-data-set.ts

```diff
@@ -0,0 +1,151 @@
+import { types, Instance } from "mobx-state-tree";
+
+export interface Color {
+  cssName: string;
+  name: string;
+  hex: string;
+}
+
+interface XYPoint {
+  x: number;
+  y: number;
+}
+// As defined in vars.sass, no easy way to get these variables in here
+// so reluctantly duplicating the definitions.
+export const ChartColors: Color[] = [
+  // bars
+  { name: "blue",   cssName: "color7",  hex: "#00aae2"},
+  { name: "orange", cssName: "color8",  hex: "#e98c42"},
+  { name: "purple", cssName: "color9",  hex: "#bd599d"},
+  { name: "green",  cssName: "color10", hex: "#49b860"},
+  // backgrounds
+  { name: "sage",   cssName: "color1",  hex: "#c9dab7"},
+  { name: "rust",   cssName: "color2",  hex: "#e0b1b3"},
+  { name: "cloud",  cssName: "color3",  hex: "#bfbfbf"},
+  { name: "gold",   cssName: "color4",  hex: "#e7e4b3"},
+  { name: "terra",  cssName: "color5",  hex: "#ecd8a5"},
+  { name: "sky",    cssName: "color6",  hex: "#a4cde9" }
+];
+
+function hexToRgb(hex: string) {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+  return result ? {
+    r: parseInt(result[1], 16),
+    g: parseInt(result[2], 16),
+    b: parseInt(result[3], 16)
+  } : null;
+}
+
+function timeSeriesSort(a: XYPoint, b: XYPoint) {
+  if (a.x < b.x) {
+    return -1;
+  }
+  if (a.x > b.x) {
+    return 1;
+  }
+  return 0;
+}
+
+export const DataPoint = types
+  .model("DataPoint", {
+    label: types.string,
+    a1: types.number,
+    a2: types.number
+  });
+
+export const ChartDataSetModel = types
+  .model("ChartDataSet", {
+    name: types.string,
+    data: types.array(DataPoint),
+    color: types.string
+  })
+  .views(self => ({
+    // labels for a data point - essential for a bar graph, optional for a line
+    get dataLabels() {
+      return self.data.map(p => p.label);
+    },
+    // Axis 1 data, for a line will be point x value, for bar will be quantity
+    get dataA1() {
+      return self.data.map(p => p.a1);
+    },
+    // Axis 2 data for a line will be y value, for a bar will be label
+    get dataA2() {
+      if (self.data.length > 0 && self.data[0].a2) {
+        return self.data.map(p => p.a2);
+      } else {
+        return self.data.map(p => p.label);
+      }
+    },
+    // Determine minimum and maximum values on each axis
+    get maxA1(): number | undefined {
+      return Math.max(...self.data.map(p => p.a1));
```

Comment:
Note that if there is no data, `Math.max(...[])` is `-Infinity`, so the graph won't render. (Same with the other calculations below.) Maybe graphs should be required to have a default for when there is no data?